modules/metastorage-server/src/test/java/org/apache/ignite/internal/metastorage/server/AbstractKeyValueStorageTest.java

```diff
@@ -2043,6 +2043,43 @@ public void rangeCursor() {
         }
     }
 
+    @Test
+    public void rangeCursorSkippingTombstones() {
+        byte[] key1 = key(1);
+        byte[] val1 = keyValue(1, 1);
+
+        byte[] key2 = key(2);
+        byte[] val2 = keyValue(2, 2);
+
+        assertEquals(0, storage.revision());
+        assertEquals(0, storage.updateCounter());
+
+        storage.put(key1, val1);
+
+        assertEquals(1, storage.revision());
+        assertEquals(1, storage.updateCounter());
+
+        storage.remove(key1);
+
+        assertEquals(2, storage.revision());
+        assertEquals(2, storage.updateCounter());
+
+        storage.put(key2, val2);
+
+        assertEquals(3, storage.revision());
+        assertEquals(3, storage.updateCounter());
+
+        // Range that includes tombstones.
+        Cursor<Entry> cur = storage.range(key1, null, true);
+
+        assertEquals(2, cur.stream().count());
```

Comment:
I'd rather add extra check not only for size, but for actual returned data set. Otherwise given test will pass if, for example, only tombstone is returned as a result for `storage.range(key1, null, false);`

Reply:
done